### PR TITLE
Inconsistent Hashing (Take 2)

### PR DESCRIPTION
**Pull request by nadnoslen**
Sorry for my previous commit and pull request that did not address tests.  In my haste to get this working in my project I left the testing behind...my bad.

### What Is Broke

What this pull request does is properly read the files in as a buffer so that they are hashed properly.  Forcing (or defaulting to) an encoding of "UTF-8" will create inconsistent hashing on many binary files (e.g. png, jpg, etc.).

#### How Do We Know It Is Broke
Here's the best way to see what I mean; go to the head of the `rickharrison/broccoli-asset-rev` repo and execute the following commands:

    $ md5 tests/fixtures/basic/input/images/sample.png 
    MD5 (tests/fixtures/basic/input/images/sample.png) = 1f6b78f1b4667adc7e397f7bf94041ab

Notice the MD5 value is `1f6b78f1b4667adc7e397f7bf94041ab` for the input file `sample.png`.  In our output directory, that we are asserting against, the sample.png is expected to hash to `b7caf246e908cd00d8011f370e3dd992`.  That is not correct as my command-line MD5 shows.

### What Fixes The Problem

This was all a result of the incorrect hashing from the `fs.readFileSync(..., {encoding: 'UTF-8'});`.  This pull request fixes this problem.

Cheers.

### CODE REVIEW

File: lib/fingerprint.js [hunk 1]
```diff
@@ -68,7 +68,7 @@ Fingerprint.prototype.getDestFilePath = function (relativePath) {
     }
 
     var tmpPath = path.join(this._srcDir, relativePath);
-    var file = fs.readFileSync(tmpPath, { encoding: 'utf8' });
+    var file = fs.readFileSync(tmpPath, { encoding: null });
```
Comment (nadnoslen) [inline lib/fingerprint.js]: Most important part of the change.  Force a buffer back from the `fs.readFileSync(...)` by passing the `null` encoding.

File: tests/fixtures/basic/output/styles-daff78e51cc77ea97e75a7fd9c5f6fae.css [hunk 2]
```diff
@@ -1,11 +1,11 @@
 .sample-img {
   width: 50px;
   height: 50px;
-  background-image: url(images/sample-b7caf246e908cd00d8011f370e3dd992.png);
+  background-image: url(images/sample-1f6b78f1b4667adc7e397f7bf94041ab.png);
```
Comment (nadnoslen) [inline tests/fixtures/basic/output/styles-daff78e51cc77ea97e75a7fd9c5f6fae.css]: I've simply updated the affected fixture output files to use the appropriate MD5 values.

**Comment by rickharrison** [comment]
Can you squash into a single commit please

**Comment by nadnoslen** [comment]
Sorry about that.  All in one commit now.  Cheers.

**Comment by rickharrison** [comment]
Thanks for catching this. Much appreciated.

**Pull request merged**